torchrl/collectors/collectors.py

```diff
@@ -841,12 +841,9 @@ def set_seed(self, seed: int) -> int:
         _check_for_faulty_process(self.procs)
         for idx in range(self.num_workers):
             self.pipes[idx].send((seed, "seed"))
-            new_seed, msg = self.pipes[idx].recv()
+            seed, msg = self.pipes[idx].recv()
             if msg != "seeded":
                 raise RuntimeError(f"Expected msg='seeded', got {msg}")
-            seed = new_seed
```

Comment:
here we increment the seed by 1, such that all sub-collectors have a different initial seed

Do we still do that?
Don't we need to generate a new seed based on the previous one?

Reply:
This Design
1. Collection sends a root seed to ParallelEnv.
2. ParallelEnv generates seed sequence excl. the root seed and seed to Env.
3. ParallelEnv returns the last used seed.
4. Collection receives a last used seed from ParallelEnv. This value is now a new root seed and feeded into the next ParallelEnv.

Pros
1. Collection does not know about the seeding algorithm. 
Cons
1. The root seed is not used. But do we need to know about it? We can always restore the exact sequence from default_rng function.

Alternative(Old) Design
1. Collection sends a root seed to ParallelEnv.
2. ParallelEnv generates seed sequence incl. the root seed and seed to Env.
3. ParallelEnv returns the last used seed.
4. Collection receives a last used seed from ParallelEnv.
5. Collection gets the next sequence from the last used seed it received from ParallelEnv. This value is now a new root seed and feeded into the next ParallelEnv.

Pros
1. Root seed is used. 
Cons
1. Collection does not need to know about the seeding algorithm.

I liked the first approach, but please let me know if you prefer the second approach!